### DISCUSSION TITLE
ctrl + s triggers onClose

### DISCUSSION THREAD

**Discussion by Vladmidir** [marcveens/react-drawio#19] [Q&A] [answered]
If you make the most basic editor page and click `ctrl + s`, this will trigger the onClose event handler.

```js
import { useEffect, useRef, useMemo, useState, useCallback } from 'react';
import { DrawIoEmbed} from 'react-drawio';


const Editor = () => {

return (
      <div style={{height: "100vh"}}>
          < DrawIoEmbed 
              onClose={(data) => console.log(data)}
              />
      </div>
);
};

export default Editor;
```

Is this the expected behaviour?

**Comment by marcveens** [comment, marked as answer]
Yes it is, for now. It was implemented like that initially, but this behaviour might change with a next major change, so that the functionality is split up.

What might help you, is that an extra parameter has been added to the onClose response object, so you can detect what triggered the onChange event. If you press ctrl +s, this is the response:
```
{
    "event": "exit",
    "modified": false,
    "parentEvent": "save"
}
```

If the exit button is clicked manually, this is the response:

```
{
    "event": "exit",
    "modified": false
}
```

The `parentEvent` can be used to change behaviour if necessary

**Reply by Vladmidir** [reply]
Thank you! I did not realize, but I was using an older version of the library, which was missing the "parentEvent" property. I've updated to the newer version and got everything to work!
